MetaMask/metamask-extension
# feat: add new authentication flow

## **Description**

This PR relates to the integration of new Authentication and User Storage controllers within the extension.

Specifically, the PR implements:
- Logic for integrating Redux with the created controllers
- Necessary hooks to manage actions and their associated loading and error states
- Integration with the onboarding process
- Integration with settings pages where necessary

At the end of the onboarding process, the user will be able to:
- Choose to use MetaMetrics
- Use the Profile Syncing feature (enabling UserStorage) or, through the advanced settings page, choose not to use this feature

If the user chooses to use MetaMetrics or enable Profile Syncing, the user must be authenticated.

Only if the user chooses not to use MetaMetrics and not to enable Profile Syncing can they remain unauthenticated.

Currently, the authentication process continues to use the locally created MetaMetrics ID.

The PR also includes components that allow the user, if they wish, to disable MetaMetrics or Profile Syncing at any time.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24028?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Follow the onboarding process
2. Check the settings page to turn on/off metametrics and profile syncing

### **After**

This is the new onboarding flow:

https://www.figma.com/file/c7GgNw2kScGrVyRGAPhwEd/Notifications?type=design&node-id=1498-68094&mode=design&t=sFvZbmQXNQaBxs6M-4

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
